### DISCUSSION TITLE
Specify Rspamd version

### DISCUSSION THREAD

**Discussion by cfis** [docker-mailserver/docker-mailserver#3922] [Q&A] [answered]
When using the Helm chart, the docker-mailserver pod name changes with each update/reinstall because the host name changes (ie,the pod name changes). This causes Rsampd to lose its history because it uses a redis key to store the history which is based on the host name.

I submitted an issue to rsampd - see https://github.com/rspamd/rspamd/pull/4801. They were nice enough to fix this by allowing the redis variable name to be set via a template. This change was released in version 3.8.2 - see https://github.com/rspamd/rspamd/releases/tag/3.8.2.

I would like to be able to use this new version of rspamd and also be able to set history key_prefix.  

Thanks!

**Comment by polarathene** [comment]
> I would like to be able to use this new version of rspamd and also be able to set history key_prefix.

That is managed by @georglauterbach , AFAIK we use packages from official Rspamd PPA:

https://github.com/docker-mailserver/docker-mailserver/blob/899b644a04cfbfed10d03d67a6864a2ef988ee32/target/scripts/build/packages.sh#L166-L179

There might be some changes there that you have to wait for until v14. Doesn't seem like we have any version pinning in place so I'm not sure how @georglauterbach manages that. DMS doesn't update existing tagged releases when published, so they may not always be equivalent to a local build.

Perhaps try a local build with `:edge`?

---

> docker-mailserver pod name changes with each update/reinstall because the host name changes (ie,the pod name changes). This causes Rsampd to lose its history because it uses a redis key to store the history which is based on the host name.

That is an example of the concerns with plans for https://github.com/docker-mailserver/docker-mailserver/issues/3520

It is not always obvious or clear when software is relying on the hostname implicitly for config/operation unfortunately. Thanks for identifying another one and getting that resolved upstream! 🙏

**Comment by georglauterbach** [comment, marked as answer]
When the changes have been published in the PPA, theoretically, the next build for `:edge` should catch the latest version of Rspamd (note: it's not _Rsampd_ but _Rspamd_ 😄). Then, it should be an easy fix, just adding the file in `target/rspamd/local.d/history_redis.conf` with contents

```ini
key_prefix = "rs_history{{COMPRESS}}";
```

The file should be created in this function:

https://github.com/docker-mailserver/docker-mailserver/blob/d227d6dc7368274fa5c2b206e77b33274b4569bc/target/scripts/startup/setup.d/security/rspamd.sh#L120-L146

Because there might exist users who use a dedicated Redis instance. I.e., we need a second HERDOC:

```bash
     cat >"${RSPAMD_LOCAL_D}/history_redis.conf" << "EOF" 
# documentation: https://rspamd.com/doc/modules/history_redis.html

key_prefix = "rs_history{{COMPRESS}}";
  
EOF
```

I can provide a PR for that.
